File: client/java/src/stm/Response.java
package stm;

import org.omg.CORBA.PUBLIC_MEMBER;

/**
 *
 * Created by xpwu on 2016/11/29.
 */
class Response {
  enum Status {
    Success(0), Failed(1);

    private int code_ = 0;
    private Status(int code) {
      this.code_ = code;
    }
  }

  long reqID;
  byte[] data;
  Status status;

  static public Response defaultResponse(){
    Response res = new Response();
    res.reqID = 0;
    res.status = Status.Success;
    res.data = null;

    return res;
  }
}

File: client/wxapp/stm/response.js
/**
 * Created by xpwu on 2016/12/1.
 */


// (function (ns) {
//   "use strict";

  // var stm = ns.stm = ns.stm || {};

  var state = {Success: 0, Failed: 1};

  function Response() {
    /**
     *
     * @type {number}
     */
    this.reqID = 0;
    /**
     *
     * @type {number}
     */
    this.state = state.Success;
    /**
     *
     * @type {ArrayBuffer}
     */
    this.data = null;
  }

  // stm.Response = Response;
  // stm.Response.State = state;

  module.exports.Response = Response;
  module.exports.Response.State = state;

// })(this);